src/components/particle-emitter.js

```diff
@@ -32,22 +35,37 @@ AFRAME.registerComponent("particle-emitter", {
     this.updateParticles = false;
   },
 
+  async setTexture(src, resolve) {
+    let accessibleUrl = src;
+
+    if (resolve) {
+      const result = await resolveUrl(src);
+      let canonicalUrl = result.origin;
+
+      // handle protocol relative urls
+      if (canonicalUrl.startsWith("//")) {
+        canonicalUrl = location.protocol + canonicalUrl;
+      }
+
+      // todo: we don't need to proxy for many things if the canonical URL has permissive CORS headers
+      accessibleUrl = proxiedUrlFor(canonicalUrl, null);
+    }
+
+    const texture = new THREE.Texture();
+
+    await textureLoader.loadTextureAsync(texture, accessibleUrl);
+
+    this.particleEmitter.material.uniforms.map.value = texture;
```

Comment:
Can particle emitters change src? If so we need to check here that src didn't change while the request was in flight.

Reply:
I'd like to make it so they could since the change is pretty simple. I'll add that, thanks 👍